burgle.lic

```diff
@@ -112,22 +201,193 @@ class Burgle
+          message("Set to cycle, but could only find rope.")    #warn user that they are missing an entry method. cycle requires presence of both to work right
+        end
+        @entry_type = "rope"
+      else
+        @entry_type = "lockpick"
+      end
+    else
+      message("Invalid priority type.")
+      exit
+    end
+  end
+
+  #Should already have checked for entry type, now get it (right before buff/robbing), but this validates anyway to be safe
+  def get_entry(entry_type)
+    case entry_type
+    when /rope/i
+      if bput('get my heavy rope', 'You get', 'You are already holding' ,'What were you') =~ /What were you/
+        message("Couldn't find entry item: heavy rope")
+        exit
+      end
+    when /lockpick/i
+      if @use_lockpick_ring
+        return
+      elsif bput('get my lockpick', 'You get', 'You are already holding' ,'What were you') =~ /What were you/
+        message("Couldn't find entry item: lockpick")
+        exit
+      end
+    end
+  end
+
+  #direction is the direction you came from to get to this room
+  def rob_the_place(direction = nil)
+    #empty array for directions you have already visited
+    visited = []
+    #if you came from a previous room, add it to the array so you don't loop
+    visited.push(direction) unless direction == nil
+
+    #get the room type, which is used to get searchable objects
+    if room_match = XMLData.room_title.match(/\[\[Someone Else's Home, (?<room>.*)\]\]/)
+      current_room = room_match[:room]
+      #search the lootable objects unless the room type has been blacklisted
+      search_for_loot(@room_searchable_objects_map[current_room]) unless @burgle_settings['room_blacklist'].include?(current_room)
+      #while you still have other rooms from this room to visit, you haven't exhausted your search count, and you haven't seen footsteps in safe_mood:
+      while XMLData.room_exits.count > visited.count and @search_count < @burgle_settings['max_search_count']  && !(Flags['burgle-footsteps']) do
+        #pick a new room to go to
+        newdir = XMLData.room_exits.reject { |exit| visited.include?(exit) }.sample
+        #store that you're going to it
+        visited.push(newdir)
+        #store how to back out
+        reverse = @reverse_direction_map[newdir]
+        #go to the picked room
+        burgle_move(newdir)
+        #loot the new room, unless you've seen footsteps in safe_mode
+        rob_the_place(reverse) unless Flags['burgle-footsteps']
+        #backup to the original room
+        burgle_move(reverse)
+      end
+    end
   end
 
   def search_for_loot(target)
-    if /It looks valuable/ =~ bput("search #{target}",'It looks valuable','Roundtime','I could not')
-      if @burgle_settings['loot'] == "drop"
-        drop_item(DRC.right_hand)
+    #don't hide/search if you've seen footsteps and are in safe_mode
+    return if Flags['burgle-footsteps']
+    #ensure you're in hiding or invis before looting -> only try once because of timer
```

Comment:
❔ If you're only going to try once, why have a loop?

Reply:
Legacy from just c/p from the same code prior to the burgle call.  I had thought about either putting in a value to let users pick how many times it tried, but realized that's probably a bad idea.  

I also thought about updating hide? to specify a number of times to try before giving up, but hide? is more complex than I originally thought.